packages/tests/tests/filter.node.spec.ts

```diff
@@ -11,203 +13,961 @@ import { createLightNode } from "@waku/sdk";
+  );
+  const decoders = contentTopics.map((topic) => createDecoder(topic));
+  return {
+    contentTopics,
+    encoders,
+    decoders
+  };
+}
+
+// Utility to validate errors related to pings in the subscription.
+async function validatePingError(
+  subscription: IFilterSubscription
+): Promise<void> {
+  try {
+    await subscription.ping();
+    throw new Error(
+      "Ping was successful but was expected to fail with a specific error."
+    );
+  } catch (err) {
+    if (
+      err instanceof Error &&
+      err.message.includes("peer has no subscriptions")
+    ) {
+      return;
+    } else {
+      throw err;
+    }
+  }
+}
+
+describe("Waku Filter: V2", function () {
+  // Set the timeout for all tests in this suite. Can be overwritten at test level
+  this.timeout(10000);
   let waku: LightNode;
   let nwaku: NimGoNode;
-
+  let nwaku2: NimGoNode;
   let subscription: IFilterSubscription;
+  let messageCollector: MessageCollector;
 
-  afterEach(async function () {
+  this.afterEach(async function () {
     !!nwaku &&
       nwaku.stop().catch((e) => console.log("Nwaku failed to stop", e));
+    !!nwaku2 &&
+      nwaku2.stop().catch((e) => console.log("Nwaku2 failed to stop", e));
     !!waku && waku.stop().catch((e) => console.log("Waku failed to stop", e));
   });
 
-  beforeEach(async function () {
+  this.beforeEach(async function () {
     this.timeout(15000);
-    nwaku = new NimGoNode(makeLogFileName(this));
-    await nwaku.start({
-      filter: true,
-      lightpush: true,
-      relay: true
-    });
-    waku = await createLightNode({
-      staticNoiseKey: NOISE_KEY_1,
-      libp2p: { addresses: { listen: ["/ip4/0.0.0.0/tcp/0/ws"] } }
-    });
-    await waku.start();
+
+    // Setup before each test to initialize nodes and message collector.
+    // Sometimes the node setup fails, when that happens we retry it max 3 times.
+    await pRetry(
+      async () => {
+        try {
+          nwaku = new NimGoNode(makeLogFileName(this));
+          await nwaku.start({
+            filter: true,
+            lightpush: true,
+            relay: true
+          });
+        } catch (error) {
+          console.log("nwaku node failed to start:", error);
+          throw error;
+        }
+      },
+      { retries: 3 }
+    );
+
+    try {
+      waku = await createLightNode({
+        staticNoiseKey: NOISE_KEY_1,
+        libp2p: { addresses: { listen: ["/ip4/0.0.0.0/tcp/0/ws"] } }
+      });
+      await waku.start();
+    } catch (error) {
+      console.log("jswaku node failed to start:", error);
+    }
     await waku.dial(await nwaku.getMultiaddrWithId());
     await waitForRemotePeer(waku, [Protocols.Filter, Protocols.LightPush]);
     subscription = await waku.filter.createSubscription();
+    messageCollector = new MessageCollector();
+  });
+
+  it("Subscribe and receive messages via lightPush", async function () {
+    // Subscribe to the content topic and set callback for received messages.
```

Comment:
It's good to have comments but to me it seems we can drop some of them sicne it is clear what's happening by the code.

Reply:
Agreed, will remove some of them :)